TINKERPOP-1026 BVLP should store vertex IDs as String

dkuppitz opened the pull request:
https://issues.apache.org/jira/browse/TINKERPOP-1026

BLVP (in particular `IncrementalBulkLoader`) now stores the `toString()` version of vertex IDs in the temporary `bulkLoader.vertex.id` property. I also added another `BulkLoader` implementation (`OneTimeBulkLoader`), to a) be able to add more test cases and b) verify that the `toString()` update doesn't interfere with something else.

Successfully ran `mvn clean install` and integration tests.

VOTE: +1

spmallette commented:
I assume that `OneTimeBulkLoader` is something users might use?  or was it really just for those two internal use cases you listed there?  If it's just internal, perhaps you shouldn't make the class visible to people?  

or if it is something that users might use, do you need to update the reference docs to discuss how it is used?  maybe some javadocs is enough? both?

dkuppitz commented:
Updated.

* added notes in the reference docs
* made one BLVP sample code snippet use the new `OneTimeBulkLoader`
* verified that `process-docs.sh` still works
* added javadocs for `OneTimeBulkLoader`

spmallette commented:
looks like you angered the travis gods. doesn't look like the error is related to your changes though.

dkuppitz commented:
Hm, worked for me locally. Can we somehow retry Travis builds w/o doing another commit?

spmallette commented:
Maybe push an empty commit? 

```text
git commit --allow-empty
```

spmallette commented on docs/src/reference/the-graphcomputer.asciidoc:
```diff
@@ -364,10 +364,11 @@ It's recommended to tune this property for the target graph and not use the defa
-has to be used if the default behavior is not sufficient.
+NOTE: `BulkLoaderVertexProgram` uses the `IncrementalBulkLoader` by default. The other option is the `OneTimeBulkLoader`,
+which doesn't store any temporary IDs in the `writeGraph` and thus should only be used for initial bulk loads. Both
+implementations should cover the mahority of use-cases, but have a limitation though: They don't support multi-valued
```
spelling mistake - should be "majority" :smile:

spmallette commented:
`mvn clean install` is good to go and code/docs look good:

VOTE +1

twilmes commented:
Tests and code look good.

VOTE +1
